src/main/java/cr0s/warpdrive/config/structures/Schematic.java

```diff
@@ -1,38 +1,75 @@
 package cr0s.warpdrive.config.structures;
 
 import cr0s.warpdrive.WarpDrive;
-import cr0s.warpdrive.config.Filler;
-import cr0s.warpdrive.config.GenericSet;
-import cr0s.warpdrive.config.InvalidXmlException;
-import cr0s.warpdrive.config.Loot;
-import cr0s.warpdrive.config.WarpDriveConfig;
-import cr0s.warpdrive.config.XmlFileManager;
-
-import javax.annotation.Nonnull;
-import java.util.List;
-import java.util.Random;
-
+import cr0s.warpdrive.config.*;
+import cr0s.warpdrive.data.JumpBlock;
 import net.minecraft.block.Block;
+import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
-
 import org.w3c.dom.Element;
 
+import javax.annotation.Nonnull;
+import java.util.*;
+
+import static cr0s.warpdrive.config.WarpDriveConfig.LOGGING_WORLD_GENERATION;
+
 public class Schematic extends AbstractStructure {
 	
-	protected String filename;
+	protected HashMap<String, Integer> fileNames;
 	protected Replacement[] replacements;
 	protected Insertion[] insertions;
 	
 	public Schematic(final String group, final String name) {
 		super(group, name);
 	}
 	
+	public String getRandomFileName(Random random) {
+		int totalWeight = 0;
+		for (int entry : fileNames.values()) {
```

Comment:
In Java collection classes, entry means a tuple of index and value, here we're only getting value. Consider renaming, for example "weight"?